File: read_excel.js
const XLSX = require('xlsx');
const fs = require('fs');

const filePath = 'C:\\Users\\ADMIN\\Downloads\\Active Accounts 12-3-2025 6-52-25 PM.xlsx';

try {
    const workbook = XLSX.readFile(filePath);
    const sheetName = workbook.SheetNames[0];
    const sheet = workbook.Sheets[sheetName];

    // Convert to JSON with raw values to preserve dates if possible, or formatted strings
    const data = XLSX.utils.sheet_to_json(sheet, { raw: false, defval: '---' });

    fs.writeFileSync('d:\\antigravity\\excel_data.json', JSON.stringify(data, null, 2), 'utf8');
    console.log('Data written to excel_data.json');
} catch (e) {
    console.error('Error reading file:', e);
}

File: read_excel.py
import pandas as pd
import json
import sys

file_path = r"C:\Users\ADMIN\Downloads\Active Accounts 12-3-2025 6-52-25 PM.xlsx"

try:
    # Read the excel file
    df = pd.read_excel(file_path)
    
    # Convert datetime objects to string format
    for col in df.columns:
        if pd.api.types.is_datetime64_any_dtype(df[col]):
            df[col] = df[col].dt.strftime('%m/%d/%Y %I:%M %p')
            
    # Replace NaN with empty string or dashes
    df = df.fillna('---')
    
    # Convert to list of dicts
    data = df.to_dict(orient='records')
    
    # Print as JSON
    print(json.dumps(data, indent=2))

except Exception as e:
    print(f"Error: {e}")
